scalafix-tests/unit/src/test/scala-2.12/scalafix/tests/cli/ScalafixImplSuite.scala

```diff
@@ -0,0 +1,192 @@
+package scalafix.tests.cli
+
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.FileOutputStream
+import java.io.PrintStream
+import java.net.URLClassLoader
+import java.nio.channels.Channels
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import org.scalatest.FunSuite
+import scalafix.Versions
+import scalafix.{interfaces => i}
+import scala.collection.JavaConverters._
+import scala.meta.internal.semanticdb.scalac.SemanticdbPlugin
+import scala.meta.io.AbsolutePath
+import scala.meta.io.Classpath
+import scalafix.interfaces.ScalafixDiagnostic
+import scalafix.interfaces.ScalafixException
+import scalafix.interfaces.ScalafixMainCallback
+import scalafix.interfaces.ScalafixMainMode
+import scalafix.internal.reflect.ClasspathOps
+import scalafix.internal.reflect.RuleCompiler
+import scalafix.internal.sbt.ScalafixJarFetcher
+import scalafix.testkit.DiffAssertions
+
+class ScalafixImplSuite extends FunSuite with DiffAssertions {
+  def semanticdbPluginPath(): String = {
```

Comment:
Can we get this from BuildInfo?

Reply:
BuildInfo doesn't robustly support task generators, we could make it a resource generator but that would make the test harder to read IMO. I inlined it here since that makes the test more self-contained (instead of magically pulling in info from the build) and avoids the need to regenerate this information on every compile/run instead of only when we run `ScalafixImplSuite`.